elements/lisk-chain/src/data_access/storage.ts

```diff
@@ -47,21 +52,17 @@ export class Storage {
 		Block headers
 	*/
 	public async getBlockHeaderByID(id: string): Promise<BlockHeaderJSON> {
-		const block = await this._db.get<BlockHeaderJSON>(
-			`${DB_KEY_BLOCKS_ID}:${id}`,
-		);
-		return block;
+		const blockBuffer = await this._db.get(`${DB_KEY_BLOCKS_ID}:${id}`);
+		return JSON.parse(blockBuffer.toString('utf8')) as BlockHeaderJSON;
 	}
 
 	public async getBlockHeadersByIDs(
 		arrayOfBlockIds: ReadonlyArray<string>,
 	): Promise<BlockHeaderJSON[]> {
 		const blocks = [];
 		for (const id of arrayOfBlockIds) {
-			const block = await this._db.get<BlockHeaderJSON>(
-				`${DB_KEY_BLOCKS_ID}:${id}`,
-			);
-			blocks.push(block);
+			const block = await this._db.get(`${DB_KEY_BLOCKS_ID}:${id}`);
+			blocks.push(JSON.parse(block.toString('utf8')) as BlockHeaderJSON);
```

Comment:
I am concerned about the performance of `JSON.parse` on such an important function here. If such calls always require to convert the buffer to string and then parse the string. Then it does not make sense storing the buffer in the first place.

Reply:
Same as above, this will be replaced by the lisk-codec